autoload/neomake/makers/ft/haskell.vim

```diff
@@ -73,9 +73,12 @@ function! neomake#makers#ft#haskell#hdevtools() abort
     if !exists('s:uses_cabal')
         let s:uses_cabal = 0
         if executable('stack')
-            let rootdir = systemlist('stack --verbosity silent path --project-root')[0]
-            if empty(glob(rootdir . '/*.cabal'))
-                let s:uses_cabal = 1
+            let output = neomake#utils#systemlist(['stack', '--verbosity silent', 'path', '--project-root'])
+            if !empty(output)
+                let rootdir = output[0]
+                if empty(glob(rootdir . '/*.cabal'))
+                    let s:uses_cabal = 1
+                endif
```

Comment:
Yah, just checked with a couple Cabal and non-Cabal projects that hdevtools if the maker is just reduced to:

````
function! neomake#makers#ft#haskell#hdevtools() abort
    let params = {
        \ 'exe': 'hdevtools',
        \ 'args': ['check', '-g-Wall'],
        \ 'mapexpr': s:CleanUpSpaceAndBackticks(),
        \ 'errorformat':
            \ '%-Z %#,'.
            \ '%W%f:%l:%v: Warning: %m,'.
            \ '%W%f:%l:%v: Warning:,'.
            \ '%E%f:%l:%v: %m,'.
            \ '%E%>%f:%l:%v:,'.
            \ '%+C  %#%m,'.
            \ '%W%>%f:%l:%v:,'.
            \ '%+C  %#%tarning: %m,'
        \ }
    return params
endfunction
````